week3/homework/src/myPackage.js

```diff
@@ -0,0 +1,52 @@
+const fs = require("fs");
+const bodyParser = require("body-parser");
+
+function readToDo(request, response) {
+  fs.readFile("./todos.json", "utf8", (error, todoList) => {
+    if (error) {
+      console.log(error);
+      response.status(500).send("some problem.");
+    } else {
+      const index = request.params.id;
+      response.json(JSON.parse(todoList)[index - 1]);
+      console.log(todoList);
+    }
+  });
+}
+
+function clearToDos(response) {
+  fs.truncate("./todos.json", 0, (error, todoList) => {
+    if (error) {
+      console.log(error);
+      response.status(500).send("Problem Occured.");
+    } else {
+      console.log("Todo list cleared.");
+      response.json();
+    }
+  });
+}
+
+function markAsDoneOrNotDone(request, response, boolean) {
+  fs.readFile("./todos.json", "utf8", (error, data) => {
+    if (error) {
+      console.log(error);
+      response.status(500).send("some problem.");
+    } else {
+      const index = request.params.id;
+      let newData = JSON.parse(data);
+      newData[index - 1].done = boolean;
```

Comment:
You need to check the ID provided by user, it might be out of range ( > Todo-List-Size or zero or negative).

Reply:
fixed it.